### PR TITLE
Annotated function

### PR DESCRIPTION
**Pull request by hkaiser**
Adding means to annotate functions to be executed by HPX

- makes annotation string available in the debugger
- integrates with VTune and APEX

### CODE REVIEW

File: hpx/runtime/actions/basic_action.hpp [hunk 1]
```diff
@@ -807,6 +836,23 @@ namespace hpx { namespace serialization
 #define HPX_DEFINE_GET_ACTION_NAME(action)                                    \
     HPX_DEFINE_GET_ACTION_NAME_(action, action)                               \
 /**/
+#if defined(HPX_HAVE_ITTNOTIFY) && !defined(HPX_HAVE_APEX)
```
Comment (sithhell) [inline hpx/runtime/actions/basic_action.hpp]: This looks like a leftover from the previous ITT branch, could you rebase this PR onto current master, please?

Comment (hkaiser) [inline hpx/runtime/actions/basic_action.hpp]: Right, I need to re-integrate the changes from the itt_notify branch to here. Rebasing onto master will not help at this point.

**Comment by sithhell** [comment]
I merged the branch to master this morning.

Am 03.02.2017 1:28 nachm. schrieb "Hartmut Kaiser" <notifications@github.com

File: hpx/util/annotated_function.hpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            std::size_t get_function_address() const
+            {
+                return traits::get_function_address<
+                        typename util::decay_unwrap<F>::type
+                    >::call(f_);
+            }
+
+            char const* get_function_annotation() const
+            {
+                return name_ ? name_ : "<unknown>";
```
Comment (sithhell) [inline hpx/util/annotated_function.hpp]: Can we use `typeid(f_).name()` instead of `"<unknown>"` here?

Comment (hkaiser) [inline hpx/util/annotated_function.hpp]: Good point. Will do.

**Comment by hkaiser** [comment]
@sithhell your comments have been addressed

**Pull request merged**